### PR TITLE
CAMEL-12641 : Aws-sns and aws-sqs components not accepting Date type …

### PR DESCRIPTION
**Pull request by saravanakumar1987**
…message attributes

Please review this PR for https://issues.apache.org/jira/browse/CAMEL-12641.

Thanks,
Saravanakumar

### CODE REVIEW

File: components/camel-aws/src/main/java/org/apache/camel/component/aws/sqs/SqsProducer.java [hunk 1]
```diff
@@ -56,33 +57,18 @@ public void process(Exchange exchange) throws Exception {
...
-            request.setMessageGroupId(messageGroupId);
-
-            MessageDeduplicationIdStrategy messageDeduplicationIdStrategy = getEndpoint().getConfiguration().getMessageDeduplicationIdStrategy();
-            String messageDeduplicationId = messageDeduplicationIdStrategy.getMessageDeduplicationId(exchange);
-            request.setMessageDeduplicationId(messageDeduplicationId);
-
-        }
-    }
-
     private void addDelay(SendMessageRequest request, Exchange exchange) {
```
Comment (oscerd) [inline components/camel-aws/src/main/java/org/apache/camel/component/aws/sqs/SqsProducer.java]: Why are you removing this block?

Comment (saravanakumar1987) [reply components/camel-aws/src/main/java/org/apache/camel/component/aws/sqs/SqsProducer.java]: Sorry I had messed up. Please check this again.

**Comment by oscerd** [comment]
Merged on master. Thanks.

**Comment by davsclaus** [comment]
Maybe the Date should be formatted to some ISO standard so its not just a toString on a Date class.

**Comment by saravanakumar1987** [comment]
@davsclaus 
The date format can be customized to any particular standard by updating the header value before calling camel aws endpoint. And default format can be the one returned by https://docs.oracle.com/javase/8/docs/api/java/util/Date.html#toString--

**Pull request closed without merging**